library/fluid/fluid_extensions/src/main/java/org/quiltmc/qsl/fluid/mixin/EntityMixin.java

```diff
@@ -0,0 +1,204 @@
+import org.quiltmc.qsl.fluid.api.FlowableFluidExtensions;
+import org.quiltmc.qsl.fluid.impl.CustomFluidInteracting;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import javax.annotation.Nullable;
+
+
+@Mixin(Entity.class)
+public abstract class EntityMixin implements CustomFluidInteracting {
+	@Shadow
+	public float fallDistance;
+	@Shadow
+	public World world;
+	@Shadow
+	protected boolean firstUpdate;
+	@Shadow
+	@Final
+	protected RandomGenerator random;
+	@Unique
+	protected boolean inCustomFluid;
+	@Unique
+	protected boolean submergedInCustomFluid;
+	@Unique
+	protected Fluid submergedCustomFluid;
+	@Shadow
+	private BlockPos blockPos;
+
+	@Shadow
+	public abstract boolean equals(Object o);
+
+	@Shadow
+	@Nullable
+	public abstract Entity getVehicle();
+
+	@Shadow
+	public abstract boolean updateMovementInFluid(TagKey<Fluid> tag, double d);
+
+	@Shadow
+	public abstract void extinguish();
+
+	@Shadow
+	public abstract double getEyeY();
+
+	@Shadow
+	public abstract double getX();
+
+	@Shadow
+	public abstract double getZ();
+
+	@Shadow
+	public abstract BlockPos getBlockPos();
+
+	@Shadow
+	public abstract boolean isSprinting();
+
+	@Shadow
+	public abstract boolean isSwimming();
+
+	@Shadow
+	public abstract void setSwimming(boolean swimming);
+
+	@Shadow
+	public abstract boolean hasVehicle();
+
+	@Shadow
+	public abstract double getY();
+
+	@Shadow
+	public abstract void setOnFireFromLava();
+
+	@Inject(method = "baseTick", at = @At("TAIL"))
+	public void baseTick(CallbackInfo ci) {
+		this.checkCustomFluidState();
+		this.updateSubmergedInCustomFluidState();
+	}
+
+	@Override
+	public boolean quilt$isInCustomFluid() {
+		return this.inCustomFluid;
+	}
+
+	@Override
+	public boolean quilt$isSubmergedInCustomFluid() {
+		return this.submergedInCustomFluid && this.quilt$isInCustomFluid();
+	}
+
+	void checkCustomFluidState() {
+		FluidState fluidState = this.world.getFluidState(this.getBlockPos());
+		/*
+		 * Check if Player is completly submerged in the custom fluid.
+		 */
+		this.submergedInCustomFluid = this.quilt$isSubmergedInCustomFluid(fluidState.getFluid());
+
+		/*
+		 * Check if Player is swimming in custom Fluid
+		 * Check if player is not in a Boat
+		 */
+		if (fluidState.getFluid() instanceof FlowableFluidExtensions fluid &&
+				!(getVehicle() instanceof BoatEntity)) {
+
+			/*
+			 * We update the Movement in the fluid,by getting the TagKey of the fluid, via the Identifier.
+			 * The rest is nearly identical to vanilla.
+			 */
+			updateMovementInFluid(TagKey.of(Registry.FLUID_KEY,fluidState.getBuiltInRegistryHolder().getKey().get().getRegistry()), fluid.getPushStrength(fluidState, (Entity) (Object) this));
```

Comment:
This makes very little sense, nor does it seem to work from walking through the code. I need to look into this further

Reply:
it does work when goin into minecraft tho